src/App.tsx

```diff
@@ -0,0 +1,118 @@
+import React, { useEffect, useState, useCallback, useRef } from "react";
+import "./App.css";
+import { secretText, keyBetweenTime, showDataTime } from "./Common/global";
+import Header from "./components/Header";
+import Main from "./components/Main";
+import { blogType } from "./Common/global";
+
+function App() {
+  const intervalRef: { current: NodeJS.Timeout | null } = useRef(null);
+  const pressKey = useRef<string>("");
+  const startSecretType = useRef<boolean>(false);
```

Comment:
We don't need to specify `string` or `boolean` here. 
https://www.typescriptlang.org/docs/handbook/type-inference.html